cmd/tailscale/cli/up.go

```diff
@@ -451,15 +455,37 @@ func runUp(ctx context.Context, args []string) error {
 		}
 		if url := n.BrowseToURL; url != nil && printAuthURL(*url) {
 			printed = true
-			fmt.Fprintf(Stderr, "\nTo authenticate, visit:\n\n\t%s\n\n", *url)
-			if upArgs.qr {
+			if upArgs.json {
+				type fields struct {
+					AuthURL string
```

Comment:
is this omitempty too?

What is the signal that the state has changed and we're now successfully up?

Reply:
Added omitempty.

It otherwise behaves the same as tailscale up, blocking until the authentication is done. I'd expected to have the parent read from the child process until it receives the JSON, do whatever with the AuthURL and QR image, and then block on wait until the child exits.